controllers/listingController.js

```diff
@@ -0,0 +1,96 @@
+import Listing from "../models/listingSchema.js";
+import Review from "../models/reviewSchema.js";
+
+const listingIndexPage = async (req, res) => {
+    const allListings = await Listing.find({});
+    res.render("listings/index.ejs", { allListings });
+};
+
+const showNewListing = (req, res) => {
+    res.render("./listings/new.ejs");
+}
+
+const showListing = async (req, res) => {
+    let { id } = req.params;
+    const listing = await Listing.findById(id)
+        .populate({
+            path: "reviews",
+            populate: {
+                path: "author",
+            },
+        })
+        .populate("owner");
+    if (!listing) {
+        req.flash("error", "Listing doesn't exist anymore 🤷🏻‍♀️");
+        res.redirect("/listings");
+    }
+    res.render("listings/show.ejs", { listing });
+}
+
+const createNewListing = async (req, res, next) => {
+    let url = req.file.path;
+    let filename = req.file.filename;
+    console.log(url, "-------", filename);
+    const newListing = new Listing(req.body.listing);
+    newListing.owner = req.user._id;
+    newListing.image = { url, filename };
+    await newListing.save();
+    req.flash("success", "New listing added successfully ✅");
+    res.redirect("/listings");
```

Comment:
In constants folder you can have a file called `EjsRoutes` which will contain all the constants for EjsRoutes like this listings

`export const LISTING_EJS_TEMPLATE = "/listings"`

Reply:
will do